### DISCUSSION TITLE
how to get oDoH working ?

### DISCUSSION THREAD

**Discussion by trinib** [DNSCrypt/dnscrypt-proxy#2110] [Q&A] [answered]
i now realized that it only has 1 live server(public) and not oDoH .. I guess it is not working .

![image](https://user-images.githubusercontent.com/18756975/166925925-3c571a0a-9cad-41ca-9eaf-e6dbbaa27ad7.png)

here is my settings
![image](https://user-images.githubusercontent.com/18756975/166926034-b85c088d-fad8-494a-9c49-cc5db3edcf86.png)

what i did wrong ?

**Comment by jedisct1** [comment]
> `server_names = ['cloudflare']

Your configuration explicitly states that the only server you want to use is `cloudflare`, not even with ODoH.

**Comment by trinib** [comment]
i entered 

>server_names = ['cloudflare', 'odoh-cloudflare']

but shows [FATAL] No servers configured

sorry if i have it wrong

**Comment by jedisct1** [comment]
I'd suggest starting with the default, example configuration, and changing only things you need, testing individual changes every time.

For ODoH you will probably need to uncomment the section for odoh sources and relays, and add routes as in the documentation.

**Reply by trinib** [reply]
> I'd suggest starting with the default, example configuration, and changing only things you need, testing individual changes every time.
> 
> For ODoH you will probably need to uncomment the section for odoh sources and relays, and add routes as in the documentation.


thanks

**Comment by trinib** [comment, marked as answer]
@jedisct1 

I used debian arm64 [dnscrypt-proxy binaries](https://github.com/jedisct1/dnscrypt-proxy/releases/latest). and its working 

![image](https://user-images.githubusercontent.com/18756975/166971695-bf201435-916c-40d5-b422-5c07d695962c.png)

When using testing repo dnscrypt-proxy debian , adding  `odoh_servers = true`  in /etc/dnscrypt-proxy/dnscrypt-proxy.toml shows error :
```
[FATAL] Unsupported key in configuration file: [odoh_servers]
```
Is that a bug or is there another `config file` somewhere else ??

**Reply by jedisct1** [reply]
Your Debian packages are probably old and predate the existence of ODoH.
